### DISCUSSION TITLE
How to debug a Parcel build which fails to execute correctly when deployed ?

### DISCUSSION THREAD

**Discussion by j13ag0** [parcel-bundler/parcel#7803] [Q&A]
This Parcel bundled project works on the dev-server and fails in the production build. The project:

https://github.com/j13ag0/parcel-GLTFloader-test

This is a stripped down example which demonstrates a problem when useing threejs and GLTFloader to load a .glb file.

The setup works as expected using the dev-server provided by Parcel, but a deployed build fails and due to apparent issues with source maps, debugging the situation is difficult.

Here is the site as deployed by Netlify:
  https://naughty-benz-d7d084.netlify.app/

For the deployed version Chrome displays this error in dev-tools:
  GLTFLoader.js:4340 Uncaught TypeError: Failed to construct 'URL': Invalid URL
      at GLTFLoader.js:4340:24
      at main.js:16:12

When running the Parcel build locally (npm run build), no build errors are reported.

Debugging this using source doesn't seem likely given these issues in Chrome dev-tools:
    - hovering on code identified as having an error - red underline with a red 'x' - displays an error that is out of context with that code
    - attempting to view contents of variables fails - the hover state displays nothing
    - setting breakpoints also does not work - refreshing the page produces the same results - without hitting breakpoints

It looks like there could be a problem with the source maps or the way they are being used by the browser.

Any suggestions would be appreciated.

-jg-

**Comment by mischnic** [comment]
Sorry for the delay, but I've opened two issues:
- https://github.com/parcel-bundler/parcel/issues/8230
- https://github.com/parcel-bundler/parcel/issues/8229

A workaround for the error would be moving the `run()` call to the end of main.js after all the other definitions.
